server/sonar-web/src/main/js/apps/issues/component-viewer/main.js

```diff
@@ -18,186 +18,101 @@
-      'click .code-issue': 'selectIssue'
-    };
+export default Marionette.ItemView.extend({
+  template () {
+    return '<div></div>';
   },
 
   initialize (options) {
-    SourceViewer.prototype.initialize.apply(this, arguments);
-    return this.listenTo(options.app.state, 'change:selectedIndex', this.select);
+    this.selectIssue = this.selectIssue.bind(this);
+    this.listenTo(options.app.state, 'change:selectedIndex', this.select);
   },
 
-  onLoaded () {
-    SourceViewer.prototype.onLoaded.apply(this, arguments);
-    this.bindShortcuts();
-    if (this.baseIssue != null) {
-      this.baseIssue.trigger('locations', this.baseIssue);
-      this.scrollToLine(this.baseIssue.get('line'));
+  onRender () {
+    this.showViewer();
+  },
+
+  onDestroy () {
+    this.unbindShortcuts();
+    unmountComponentAtNode(this.el);
+  },
+
+  showViewer () {
+    if (!this.baseIssue) {
+      return;
     }
+
+    const componentKey = this.baseIssue.get('component');
+
+    render((
+      <WithStore>
+        <SourceViewer
+          component={componentKey}
+          displayAllIssues={true}
+          onIssueSelect={this.selectIssue}
+          selectedIssue={this.baseIssue.get('key')}/>
+      </WithStore>
+    ), this.el);
+  },
+
+  openFileByIssue (issue) {
+    this.baseIssue = issue;
+    this.selectedIssue = issue.get('key');
+    this.showViewer();
+    this.bindShortcuts();
+    this.scrollToLine(this.baseIssue.get('line'));
   },
 
   bindShortcuts () {
-    const that = this;
-    const doAction = function (action) {
-      const selectedIssueView = that.getSelectedIssueEl();
-      if (!selectedIssueView) {
-        return;
-      }
-      selectedIssueView.find('.js-issue-' + action).click();
-    };
     key('up', 'componentViewer', () => {
-      that.options.app.controller.selectPrev();
+      this.options.app.controller.selectPrev();
       return false;
     });
     key('down', 'componentViewer', () => {
-      that.options.app.controller.selectNext();
+      this.options.app.controller.selectNext();
       return false;
     });
     key('left,backspace', 'componentViewer', () => {
-      that.options.app.controller.closeComponentViewer();
+      this.options.app.controller.closeComponentViewer();
       return false;
     });
-    key('f', 'componentViewer', () => doAction('transition'));
-    key('a', 'componentViewer', () => doAction('assign'));
-    key('m', 'componentViewer', () => doAction('assign-to-me'));
-    key('p', 'componentViewer', () => doAction('plan'));
-    key('i', 'componentViewer', () => doAction('set-severity'));
-    key('c', 'componentViewer', () => doAction('comment'));
   },
 
   unbindShortcuts () {
-    return key.deleteScope('componentViewer');
-  },
-
-  onDestroy () {
-    SourceViewer.prototype.onDestroy.apply(this, arguments);
-    this.unbindScrollEvents();
-    return this.unbindShortcuts();
+    key.deleteScope('componentViewer');
   },
 
   select () {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this function to use "return" consistently. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3801)